pkg/providers/vault/client.go

```diff
@@ -24,17 +25,39 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+
 	"github.com/edgexfoundry/go-mod-secrets/pkg"
 )
 
+const (
+	AuthTypeHeader = "X-Vault-Token"
+)
+
 // Client defines the behavior for interacting with the Vault REST secret key/value store via HTTP(S).
 type Client struct {
 	HttpConfig SecretConfig
 	HttpCaller Caller
+
+	// internal member variables
+	lc logger.LoggingClient
 }
 
+// package stateful map variable to handle the case of the same caller to have
+// multiple secret clients with potentially the same tokens while renewing token
+// in the background go-routine
+var vaultTokenToCancelFunc = make(map[string]context.CancelFunc)
+
 // NewSecretClient constructs a SecretClient which communicates with Vault via HTTP(S)
-func NewSecretClient(config SecretConfig) (pkg.SecretClient, error) {
+// lc is the Edgex customized logger
+// bkgCtx is the background context that can be used to cancel or cleanup the background process when it is no longer needed
+// errChan is the error channel to receive any error from the background go-routine calls
+func NewSecretClient(config SecretConfig, lc logger.LoggingClient, bkgCtx context.Context, errChan chan<- error) (pkg.SecretClient, error) {
```

Comment:
Idiomatic usage requires context be passed as `ctx context.Context`; further that it be the first parameter in list.